includes/api/class-wc-rest-system-status-tools-controller.php

```diff
@@ -0,0 +1,434 @@
+
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $item['id'] ) );
+
+		return $response;
+	}
+
+    /**
+	 * Get the system status tools schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'system_status_tool',
+			'type'       => 'object',
+			'properties' => array(
+				'id'               => array(
+					'description'  => __( 'A unique identifier for the tool.', 'woocommerce' ),
+					'type'         => 'string',
+					'context'      => array( 'view', 'edit' ),
+					'arg_options'  => array(
+						'sanitize_callback' => 'sanitize_title',
+					),
+				),
+				'name'            => array(
+					'description'  => __( 'Tool name.', 'woocommerce' ),
+					'type'         => 'string',
+					'context'      => array( 'view', 'edit' ),
+					'arg_options'  => array(
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+				),
+				'action'            => array(
+					'description'  => __( 'What running the tool will do.', 'woocommerce' ),
+					'type'         => 'string',
+					'context'      => array( 'view', 'edit' ),
+					'arg_options'  => array(
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+				),
+				'description'      => array(
+					'description'  => __( 'Tool description.', 'woocommerce' ),
+					'type'         => 'string',
+					'context'      => array( 'view', 'edit' ),
+					'arg_options'  => array(
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+				),
+				'success'      => array(
+					'description'  => __( 'Did the tool run successfully?', 'woocommerce' ),
+					'type'         => 'boolean',
+					'context'      => array( 'edit' ),
+				),
+				'message'      => array(
+					'description'  => __( 'Tool return message.', 'woocommerce' ),
+					'type'         => 'string',
+					'context'      => array( 'edit' ),
+					'arg_options'  => array(
+						'sanitize_callback' => 'sanitize_text_field',
+					),
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param string $id
+	 * @return array
+	 */
+	protected function prepare_links( $id ) {
+		$base  = '/' . $this->namespace . '/' . $this->rest_base;
+		$links = array(
+			'item' => array(
+				'href'       => rest_url( trailingslashit( $base ) . $id ),
+				'embeddable' => true,
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Get any query params needed.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		return array(
+			'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+		);
+	}
+
+	/**
+	 * Actually executes a a tool.
+	 *
+	 * @param  string $tool
+	 * @return array
+	 */
+	public function execute_tool( $tool ) {
+		global $wpdb;
+		$ran = true;
+		switch ( $tool ) {
+			case 'clear_transients' :
```

Comment:
We should keep underscores. All our API paths are currently using `_` and these slugs will be used in the path name for actually executing the tool.